sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -250,174 +241,70 @@ export class EventHubReceiver extends LinkEntity {
+      "[%s] 'session_error' event occurred on the session of receiver '%s' with address '%s'. " +
+        "The associated error is: %O",
+      this._context.connectionId,
+      this.name,
+      this.address,
+      sessionError
+    );
 
-    if (this._onError) {
+    if (this._onError && sessionError) {
       const error = translate(sessionError);
-      logger.warning(
-        "[%s] An error occurred on the session for Receiver '%s': %O.",
-        this._context.connectionId,
-        this.name,
-        error
-      );
       logErrorStackTrace(error);
-
-      logger.verbose(
-        "[%s] Since the user did not close the receiver, " +
-          "we let the user know about it by calling the user's error handler.",
-        this._context.connectionId
-      );
       this._onError(error);
     }
   }
 
   private async _onAmqpClose(context: EventContext): Promise<void> {
     const rheaReceiver = this._receiver || context.receiver;
-    if (!rheaReceiver || rheaReceiver.isItselfClosed()) {
-      logger.verbose(
-        "[%s] 'receiver_close' event occurred on the receiver '%s' with address '%s' " +
-          "because the sdk initiated it. Hence not calling detached from the _onAmqpClose" +
-          "() handler.",
-        this._context.connectionId,
-        this.name,
-        this.address
-      );
-      return;
-    }
-
-    const amqpError = rheaReceiver.error;
-    if (amqpError) {
-      logger.verbose(
-        "[%s] 'receiver_close' event occurred for receiver '%s' with address '%s'. " +
-          "The associated error is: %O",
-        this._context.connectionId,
-        this.name,
-        this.address,
-        amqpError
-      );
-    }
-
-    if (!this.isConnecting) {
-      logger.verbose(
-        "[%s] 'receiver_close' event occurred on the receiver '%s' with address '%s' " +
-          "and the sdk did not initiate this. The receiver is not reconnecting. Hence, calling " +
-          "detached from the _onAmqpClose() handler.",
-        this._context.connectionId,
-        this.name,
-        this.address
-      );
-      await this.onDetached(amqpError);
-    } else {
-      logger.verbose(
-        "[%s] 'receiver_close' event occurred on the receiver '%s' with address '%s' " +
-          "and the sdk did not initate this. Moreover the receiver is already re-connecting. " +
-          "Hence not calling detached from the _onAmqpClose() handler.",
-        this._context.connectionId,
-        this.name,
-        this.address
-      );
+    logger.verbose(
+      "[%s] 'receiver_close' event occurred on the receiver '%s' with address '%s'. " +
+        "Value for isItselfClosed on the receiver is: '%s'.",
+      this._context.connectionId,
+      this.name,
+      this.address,
+      rheaReceiver ? rheaReceiver.isItselfClosed().toString() : undefined
+    );
+    if (rheaReceiver && !this.isConnecting) {
+      // Call close to clean up timers & other resources
+      await rheaReceiver.close();
```

Comment:
Same comment here, should we swallow/log errors from close?

Reply:
Fixed with https://github.com/Azure/azure-sdk-for-js/pull/6737/commits/06e9251b0941110ccdaa398a637d4fba818f69f1